### DISCUSSION TITLE
Perform action during runtime if test passes or fails

### DISCUSSION THREAD

**Discussion by f00dog-x300** [pytest-dev/pytest#9918] [Q&A]
Hello, 

My question is to see if there's a way to perform an action after each test function is run and depending whether it passes or fails will perform a different action. 

The closest I have ever gotten is looking at the `pytest_runtest_teardown` plugin, but I do not see a clear way where I can pull the test case pass/fail status or examples of it. I was wondering if anyone has any idea on how to do it? 

Thanks in advance.

**Comment by portyu9** [comment]
Hi @f00dog-x300 

`pytest_runtest_logreport` is the simplest hook for this.

Each test has reports for setup, call, and teardown, so if your action should depend on the result of the actual test body, check `report.when == "call"`:

```python
def pytest_runtest_logreport(report):
    if report.when != "call":
        return

    if report.passed:
        do_something_on_success()
    elif report.failed:
        do_something_on_failure()
    elif report.skipped:
        do_something_on_skip()
```

Using a fixture teardown alone is less suitable because it does not directly receive the test's `TestReport`.

One distinction to keep in mind: a successful `call` report does not necessarily mean the **entire test lifecycle** succeeded — teardown can still fail afterward. If that matters, also handle the `teardown` report.

Please mark this answer as accepted if it helped, thank you.
